src/main/java/bisq/core/dao/node/parser/TxParser.java

```diff
@@ -115,11 +115,15 @@ public TxParser(PeriodService periodService,
             // We keep the temporary opReturn type in the parsingModel object.
             checkArgument(!outputs.isEmpty(), "outputs must not be empty");
             int lastIndex = outputs.size() - 1;
-            txOutputParser.processOpReturnCandidate(outputs.get(lastIndex));
+            int lastNonOpReturnIndex = lastIndex;
+            if (txOutputParser.processOpReturnCandidate(outputs.get(lastIndex))) {
+                txOutputParser.processTxOutput(true, outputs.get(lastIndex), lastIndex);
```

Comment:
I am not 100% if that can cause problems. We can discuss tomorrow.

Reply:
From what I see there is only the lockup opreturn processing that really does anything here, and that was the one I wanted to fix. Would be good with more eyes to verify that of course.